ui/pages/confirmations/components/confirm/info/hooks/useFeeCalculations.ts

```diff
@@ -0,0 +1,135 @@
+import { GasFeeEstimates } from '@metamask/gas-fee-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { Hex } from '@metamask/utils';
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../shared/constants/common';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  multiplyHexes,
+} from '../../../../../../../shared/modules/conversion.utils';
+import { getConversionRate } from '../../../../../../ducks/metamask/metamask';
+import { useFiatFormatter } from '../../../../../../hooks/useFiatFormatter';
+import { useGasFeeEstimates } from '../../../../../../hooks/useGasFeeEstimates';
+import { getCurrentCurrency } from '../../../../../../selectors';
+import { HEX_ZERO } from '../shared/constants';
+import { getGasFeeEstimate } from './getGasFeesEstimate';
+import { useEIP1559TxFees } from './useEIP1559TxFees';
+import { useSupportsEIP1559 } from './useSupportsEIP1559';
+
+const EMPTY_FEE = '';
+const EMPTY_FEES = {
+  currentCurrencyFee: EMPTY_FEE,
+  nativeCurrencyFee: EMPTY_FEE,
+};
+
+export function useFeeCalculations(transactionMeta: TransactionMeta) {
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const conversionRate = useSelector(getConversionRate);
+  const fiatFormatter = useFiatFormatter();
+
+  const getFeesFromHex = useCallback(
+    (hexFee: string) => {
+      const nativeCurrencyFee =
+        getEthConversionFromWeiHex({
+          value: hexFee,
+          fromCurrency: EtherDenomination.GWEI,
+          numberOfDecimals: 4,
+        }) || `0 ${EtherDenomination.ETH}`;
+
+      const currentCurrencyFee = fiatFormatter(
+        Number(
+          getValueFromWeiHex({
+            value: hexFee,
+            conversionRate,
+            fromCurrency: EtherDenomination.GWEI,
+            toCurrency: currentCurrency,
+            numberOfDecimals: 2,
+          }),
+        ),
+      );
+
+      return { currentCurrencyFee, nativeCurrencyFee };
+    },
+    [conversionRate, currentCurrency, fiatFormatter],
+  );
+
+  const { maxFeePerGas, maxPriorityFeePerGas } =
+    useEIP1559TxFees(transactionMeta);
+  const { supportsEIP1559 } = useSupportsEIP1559(transactionMeta);
+  const gasEstimate = getGasFeeEstimate(transactionMeta, supportsEIP1559);
```

Comment:
Another one where being explicit with `gasFeeEstimate` could really help readability since we're dealing with both gas limits and gas fees.